hat/assets/js/apps/Iaso/components/forms/InputComponent.tsx

```diff
@@ -0,0 +1,226 @@
+import React, { useState } from 'react';
+import { Box } from '@material-ui/core';
+import {
+    TextInput,
+    PasswordInput,
+    NumberInput,
+    Radio,
+    Checkbox,
+    ArrayFieldInput,
+    SearchInput,
+    // @ts-ignore
+    translateOptions,
+    Select,
+    useSafeIntl,
+} from 'bluesquare-components';
+import MESSAGES from '../../domains/forms/messages';
+
+export type InputComponentProps = {
+    type?: string;
+    keyValue: string;
+    value?: any;
+    errors?: string[];
+    // eslint-disable-next-line no-unused-vars
+    onChange?: (key: string, value: any) => void;
+    options?: any[];
```

Comment:
Isn't that `DropDownOptions[]`

That may be annoying because if I remember well it's `DropDownOption<T>`. Then we may have to add a generic type to `InputComponent` 

The syntax is a bit exotic:

```typescript
const InputComponent = <T,> ({//props etc...
```